### PR TITLE
Extend button resizing to include height

### PR DESCRIPTION
**Pull request by zenseii**
This PR expands the button resizing function to allow for buttons with varying heights with or without varying widths.

Note: The empty vertical button had to be reduced from 121px in height to 120px for its height to be evenly divided into 5 parts when increasing the height.

This means that vertical buttons can now be taller than originally, allowing for longer words:
image
Note: In the future buttons like these on the same screen should have a shared minimum height to ensure symmetric sizes.

This is an animation showing the steps of generating a new button with increased height and width:
buttonExpansion

### CODE REVIEW

File: src/fheroes2/heroes/heroes_dialog.cpp [hunk 1]
```diff
@@ -344,9 +344,10 @@ int Heroes::OpenDialog( const bool readonly, const bool fade, const bool disable
 
     // Hero dismiss button.
     dst_pt.x = dialogRoi.x + 9;
-    dst_pt.y = dialogRoi.y + 318;
-    fheroes2::ButtonSprite buttonDismiss( dst_pt.x, dst_pt.y, fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 0 ),
-                                          fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 1 ), fheroes2::AGG::GetICN( ICN::DISMISS_HERO_DISABLED_BUTTON, 0 ) );
+    dst_pt.y = dialogRoi.y + 378;
+    fheroes2::ButtonSprite buttonDismiss( dst_pt.x, dst_pt.y - fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 0 ).height() / 2,
+                                          fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 0 ), fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 1 ),
```
Comment (Districh-ru) [inline src/fheroes2/heroes/heroes_dialog.cpp:349]: We can call `GetICN()` once for `ICN::BUTTON_VERTICAL_DISMISS, 0`:
```suggestion
    const fheroes2::Sprite & dismissPressed = fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 0 );
    fheroes2::ButtonSprite buttonDismiss( dst_pt.x, dst_pt.y - dismissPressed.height() / 2,
                                          dismissPressed, fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 1 ),
```

File: src/fheroes2/heroes/heroes_dialog.cpp [hunk 2]
```diff
@@ -356,14 +357,16 @@ int Heroes::OpenDialog( const bool readonly, const bool fade, const bool disable
         buttonDismiss.hide();
     }
     else {
-        fheroes2::addGradientShadow( fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 0 ), display, dst_pt, { -3, 5 } );
+        fheroes2::addGradientShadow( fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 0 ), display,
+                                     { dst_pt.x, dst_pt.y - fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 0 ).height() / 2 }, { -3, 5 } );
```
Comment (Districh-ru) [inline src/fheroes2/heroes/heroes_dialog.cpp:361]: The same comment as the previous one:
```suggestion
        const fheroes2::Sprite & dismissPressed = fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_DISMISS, 0 );
        fheroes2::addGradientShadow( dismissPressed, display, { dst_pt.x, dst_pt.y - dismissPressed.height() / 2 }, { -3, 5 } );
```

File: src/fheroes2/heroes/heroes_dialog.cpp [hunk 3]
```diff
@@ -356,14 +357,16 @@ int Heroes::OpenDialog( const bool readonly, const bool fade, const bool disable
...
 
     // Hero Patrol mode button (used in Editor).
-    fheroes2::ButtonSprite buttonPatrol( dst_pt.x, dst_pt.y, fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 0 ),
-                                         fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 1 ) );
+    fheroes2::ButtonSprite buttonPatrol( dst_pt.x, dst_pt.y - fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 0 ).height() / 2,
+                                         fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 0 ), fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 1 ) );
     if ( isEditor ) {
-        fheroes2::addGradientShadow( fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 0 ), display, { dialogRoi.x + 9, dialogRoi.y + 318 }, { -3, 5 } );
+        fheroes2::addGradientShadow( fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 0 ), display,
+                                     { dialogRoi.x + 9, dst_pt.y - fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 0 ).height() / 2 }, { -3, 5 } );
```
Comment (Districh-ru) [inline src/fheroes2/heroes/heroes_dialog.cpp:369]: And here too :)
```cpp
    const fheroes2::Sprite & patrolPressed = fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 0 );
    const int32_t patrolOffsetY = dst_pt.y - patrolPressed.height() / 2;
    fheroes2::ButtonSprite buttonPatrol( dst_pt.x, patrolOffsetY, patrolPressed, fheroes2::AGG::GetICN( ICN::BUTTON_VERTICAL_PATROL, 1 ) );
    if ( isEditor ) {
        fheroes2::addGradientShadow( patrolPressed, display, { dialogRoi.x + 9, patrolOffsetY }, { -3, 5 } );
```

**Review by Districh-ru — CHANGES_REQUESTED** [review]
Hi, @zenseii, I left several comments. Could you please take a look when you have time.

**Review by Districh-ru — APPROVED** [review]
👍

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , many thanks for these changes!

**Pull request merged**